WalletWasabi/Services/UpdateManager.cs

```diff
@@ -1,329 +1,286 @@
-		{
-			IEnumerable<FileSystemInfo> corruptedFiles = folder.GetFileSystemInfos().Where(file => file.Extension.Equals(".tmp"));
-			foreach (var file in corruptedFiles)
-			{
-				File.Delete(file.FullName);
-			}
-		}
+		File.Delete(filePath);
+		var httpClient = httpClientFactory.CreateClient($"{uri.Host}-installers");
+		httpClient.DefaultRequestHeaders.TryAddWithoutValidation("User-Agent", _userAgentGetter());
+		using var request = new HttpRequestMessage(HttpMethod.Get, uri);
+		var response = await httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+		response.EnsureSuccessStatusCode();
+		var contentStream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
+		await using var fileStream = new FileStream(filePath, FileMode.Create);
+		await contentStream.CopyToAsync(fileStream, cancellationToken).ConfigureAwait(false);
+		return filePath;
 	}
 
-	private void Cleanup()
+	private static async Task VerifySha256SumsFileAsync(string sha256SumsAscFilePath, string wasabiSignatureFilePath,
+		CancellationToken cancellationToken)
 	{
-		try
+		// Read the content file
+		byte[] bytes = await File.ReadAllBytesAsync(sha256SumsAscFilePath, cancellationToken).ConfigureAwait(false);
+		var computedHash = new uint256(SHA256.HashData(bytes));
+
+		// Read the signature file
+		var signatureText = await File.ReadAllTextAsync(wasabiSignatureFilePath, cancellationToken).ConfigureAwait(false);
+		var signatureBytes = Convert.FromBase64String(signatureText);
+
+		var wasabiSignature = ECDSASignature.FromDER(signatureBytes);
+
+		var pubKey = new PubKey(Constants.WasabiPubKey);
+
+		if (!pubKey.Verify(computedHash, wasabiSignature))
 		{
-			var folder = new DirectoryInfo(_installerDir);
-			if (folder.Exists)
-			{
-				Directory.Delete(_installerDir, true);
-			}
+			throw new InvalidOperationException("Invalid wasabi signature.");
 		}
-		catch (Exception exc)
+	}
+
+	private static async Task VerifyInstallerHashAsync(string installerFilePath, string expectedHash, CancellationToken cancellationToken)
+	{
+		var bytes1 = await File.ReadAllBytesAsync(installerFilePath, cancellationToken).ConfigureAwait(false);
+		var computedHash = SHA256.HashData(bytes1);
+		var downloadedHash = Convert.ToHexString(computedHash).ToLower();
+
+		if (expectedHash != downloadedHash)
 		{
-			Logger.LogError("Failed to delete installer directory.", exc);
+			throw new InvalidOperationException("Downloaded file hash doesn't match expected hash.");
 		}
 	}
 
-	public void StartInstallingNewVersion()
+	private static string GetInstallerName(Version version) =>
+		(PlatformInformation.GetOsPlatform(), RuntimeInformation.ProcessArchitecture) switch
+		{
+			(OS.Windows, _) => $"Wasabi-{version}.msi",
+			(OS.OSX, Architecture.Arm64) => $"Wasabi-{version}-arm64.dmg",
+			(OS.OSX, _) => $"Wasabi-{version}.dmg",
+			(OS.Linux, _) => $"Wasabi-{version}.msi",
+			//(OS.Linux, Architecture.X64) => $"Wasabi-{version}-linux-x64.tar.gz",
```

Comment:
Can Linux use a `.msi` installer?

Reply:
oh no sorry, it was for testing. sorry.

Reply:
oh wait, I misunderstood you question. I will check now

Reply:
Don't worry, I reverted back to the original.